src/utils.js

```diff
@@ -200,32 +220,171 @@ export const getAgeData = (birthMonth, birthYear, age, ageMonths) => {
+      flags.push('accuracyTooLow');
+    }
+
+    const isReliable = flags.filter((x) => !ignoredReliabilityFlags.includes(x)).length === 0;
+
+    return { flags, isReliable };
+  };
+}
+
+/**
+ * Tracks response times and invokes callback function when response times
+ * exceed specified thresholds.
+ *
+ * @class ResponseTimeTracker
+ * @param {Function} evaluateValidity function to be called to generate an decision on the
+ *    reliability of a run
+ * @param {Function} addEngagementFlags function passed through to update the run's
+ *  firekit object with flags tripped
+ * @param {number} minResponsesRequired The minimum number of responses required before
+ *   checking for threshold exceedance.
+ * @property {Array<number>} _responseTimes An array to store the response times.
+ * @property {Array<number>} _responses An array to store the responses
+ * (keypresses or button choices).
+ * @property {Array<number>} _correct An array to store the correctness of the responses.
+ */
+export class ValidityEvaluator {
+  constructor({
+    evaluateValidity = () => {},
+    addEngagementFlags = () => {},
+    minResponsesRequired = 0,
+  }) {
+    this.evaluateValidity = evaluateValidity;
+    this.addEngagementFlags = addEngagementFlags;
+    this.minResponsesRequired = minResponsesRequired;
+    this._responseTimes = [];
+    this._responses = [];
+    this._correct = [];
+  }
+
+  /**
+   *  @function startNewBlockValidation Called when a new block is started to reset the
+   * data arrays and update the evaluateValidity function if needed
+   * @param {Function} evaluateValidity
+   */
+  startNewBlockValidation(evaluateValidity = this.evaluateValidity) {
+    this.evaluateValidity = evaluateValidity;
+    this._responseTimes = [];
+    this._responses = [];
+    this._correct = [];
+  }
+
+  /**
+   * Updates ValidityEvaluator arrays (responseTimes, response, and accuracy) with
+   *
+   * @function addResponseData
+   * @param {number} responseTime Time it took for a user to respond to a stimulus
+   * @param {string} response  Choice that a user responded with
+   * ex: left_arrow, right_arrow, button_3
+   * @param {number} isCorrect 1 if a user answered correctly, 0if answered incorrectly
+   */
+  addResponseData(responseTime, response, isCorrect) {
+    // TODO: make response data arrays private
+    this._responseTimes.push(responseTime);
+    this._responses.push(response);
+    this._correct.push(isCorrect);
+
+    if (this._responseTimes.length >= this.minResponsesRequired) {
+      const { flags, isReliable } = this.evaluateValidity({
+        responseTimes: this._responseTimes,
+        responses: this._responses,
+        correct: this._correct,
+      });
+
+      if (flags?.length > 0) {
+        this.addEngagementFlags(flags, isReliable);
+      } else {
+        // If responseData has exceeded the threshold, mark the run as reliable
+        this.addEngagementFlags(flags, true);
```

Comment:
i wonder if simply add 
` this.addEngagementFlags(flags, isReliable);`
would be enough here as isReliable has already been updated in your line 312

Reply:
Good catch! Let me remove the redundant logic